File: alghoritms/bubble sort/1.js
function bubbleSort(array) {
  for (let i = 0; i < array.length; i++) {
    for (let j = array.length; j > i; j--) {
      if (array[j] < array[j - 1]) {
        swapItemsInList(array, j, j - 1);
      }
    }
  }
}

function swapItemsInList(list, indexA, indexB) {
  let temp = list[indexA];
  list[indexA] = list[indexB];
  list[indexB] = temp;
}

const arr = [9, 20, 3, 1, 8, 10, 11, 5];

bubbleSort(arr);

console.log(arr);

File: alghoritms/bubble sort/1.py
# coding=utf-8

"""

Алгоритм сортировки пузырьком

@:param array: лист элементов

"""


def bubble_sort(array):

    for i in range(len(array)):
        swapped = False
        for j in range(0, len(array) - i - 1):
            if array[j] > array[j + 1]:
                swapped = True
                array[j], array[j + 1] = array[j + 1], array[j]

        if not swapped:
            break


arr = [9, 20, 3, 1, 8, 10, 11, 5]

bubble_sort(arr)

print(arr)  # [1, 3, 5, 8, 9, 10, 11, 20]
